delira/training/backends/torch/trainer.py

```diff
@@ -0,0 +1,578 @@
+            train_metrics, val_metrics, lr_scheduler_cls,
+            lr_scheduler_params, gpu_ids, save_freq, optim_fn, key_mapping,
+            logging_type, logging_kwargs, fold, callbacks, start_epoch,
+            metric_keys, convert_batch_to_npy_fn, val_freq)
+
+        self._setup(network, optim_fn, optimizer_cls, optimizer_params,
+                    lr_scheduler_cls, lr_scheduler_params, gpu_ids,
+                    key_mapping, convert_batch_to_npy_fn,
+                    mixed_precision, mixed_precision_kwargs)
+
+        for key, val in kwargs.items():
+            setattr(self, key, val)
+
+    def _setup(self, network, optim_fn, optimizer_cls, optimizer_params,
+               lr_scheduler_cls, lr_scheduler_params, gpu_ids,
+               key_mapping, convert_batch_to_npy_fn, mixed_precision,
+               mixed_precision_kwargs):
+        """
+        Defines the Trainers Setup
+
+        Parameters
+        ----------
+        network : :class:`AbstractPyTorchNetwork`
+            the network to train
+        optim_fn : function
+            creates a dictionary containing all necessary optimizers
+        optimizer_cls : subclass of torch.optim.Optimizer
+            optimizer class implementing the optimization algorithm of
+            choice
+        optimizer_params : dict
+        lr_scheduler_cls : Any
+            learning rate schedule class: must implement step() method
+        lr_scheduler_params : dict
+            keyword arguments passed to lr scheduler during construction
+        gpu_ids : list
+            list containing ids of GPUs to use; if empty: use cpu instead
+        convert_batch_to_npy_fn : type
+            function converting a batch-tensor to numpy
+        mixed_precision : bool
+            whether to use mixed precision or not (False per default)
+        mixed_precision_kwargs : dict
+            additional keyword arguments for mixed precision
+
+        """
+
+        self.optimizers = optim_fn(network, optimizer_cls,
+                                   **optimizer_params)
+
+        super()._setup(network, lr_scheduler_cls, lr_scheduler_params,
+                       gpu_ids, key_mapping, convert_batch_to_npy_fn,
+                       network.prepare_batch)
+
+        # Load latest epoch file if available
+        if os.path.isdir(self.save_path):
+            latest_state_path, latest_epoch = self._search_for_prev_state(
+                self.save_path)
+
+            if latest_state_path is not None:
+
+                # if pth file does not exist, load pt file instead
+                if not os.path.isfile(latest_state_path):
+                    latest_state_path = latest_state_path[:-1]
+
+                logger.info("Attempting to load state from previous \
+                            training from %s" % latest_state_path)
+                try:
+                    self.update_state(latest_state_path)
```

Comment:
we should set `self.start_epoch = latest_epoch if` we want to resume training. check same for all other trainers.

Reply:
I added this, although this was already done by https://github.com/delira-dev/delira/blob/additional_backends/delira/training/backends/torch/trainer.py#L549 

In all other trainers this was already done